src/platform/StdSchedulerWin32.cpp

```diff
@@ -87,9 +87,8 @@ bool StdScheduler::DoScheduleProcs(int iTimeout)
 	// Iterate over the index because procedures may be added or removed during execution
 	// (If they are removed, we skip one execution, which doesn't really matter in practice)
 	auto tNow = C4TimeMilliseconds::Now();
-	for (size_t i_proc = 0u; i_proc < procs.size(); ++i_proc)
```

Comment:
I assume that there is a possibility the collection is modified in the loop (`Execute` maybe?). If that is the case, all the useful iterators (including the past-the-end iterator) will be invalidated (because `procs` is a vector).
Making it a `list` may be advantageous **if the current element isn't removed**. This however loses random-access. A textual search for `procs` shows a handful of instances (I count 3) where `operator[]` is used, each of which can be systematically changed to something not requiring random-access. One thing that might get affected by the change is the locality of reference.
Please let me know what changes to make (or revert).